File: mixin-base/src/main/java/net/yakclient/mixin/base/registry/pool/MethodLocation.java
package net.yakclient.mixin.base.registry.pool;

import java.util.Objects;

public class MethodLocation extends ClassLocation {
    private final String method;

    public MethodLocation(String cls, String method) {
        super(cls);
        this.method = method;
    }

    public static MethodLocation fromDataDest(MixinMetaData data) {
        return new MethodLocation(data.getClassTo(), data.getMethodTo());
    }

    public static MethodLocation fromDataOrigin(MixinMetaData data) {
        return new MethodLocation(data.getClassFrom(), data.getMethodFrom());
    }


    public String getMethod() {
        return method;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;
        final var that = (MethodLocation) o;
        return Objects.equals(method, that.method);
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), method);
    }
}

File: base/src/main/kotlin/net/yakclient/mixin/base/registry/pool/MethodLocation.kt
package net.yakclient.mixin.base.registry.pool

import java.util.*

open class MethodLocation(cls: String, val method: String) : ClassLocation(cls) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        if (!super.equals(other)) return false
        val that = other as MethodLocation
        return method == that.method
    }

    override fun hashCode(): Int {
        return Objects.hash(super.hashCode(), method)
    }

    companion object {
        fun fromDataDest(data: MixinMetaData): MethodLocation {
            return MethodLocation(data.classTo, data.methodTo)
        }

        fun fromDataOrigin(data: MixinMetaData): MethodLocation {
            return MethodLocation(data.classFrom, data.methodFrom)
        }
    }
}